index 8e41d5e..d4bb7de 100644
--- src/googletest.h
+++ src/googletest.h
@@ -14,6 +14,9 @@
 #include <string>
 #include <vector>

+#include <stdio.h>
+#include <stdlib.h>
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
index 14c9789..47ba36f 100644
--- src/logging_unittest.cc
+++ src/logging_unittest.cc
@@ -23,6 +23,9 @@
 #include <string>
 #include <vector>

+#include <stdio.h>
+#include <stdlib.h>
+
 #include "base/commandlineflags.h"
 #include "glog/logging.h"
 #include "glog/raw_logging.h"
index e6e3e70..0e1b6cb 100644
--- src/symbolize.cc
+++ src/symbolize.cc
@@ -414,7 +414,7 @@ class LineReader {

   char *FindLineFeed() {
     return reinterpret_cast<char *>
-      (memchr(reinterpret_cast<const void *>(bol_), '\n', eod_ - bol_));
+      (memchr(reinterpret_cast<void *>(bol_), '\n', eod_ - bol_));
   }

   bool BufferIsEmpty() {
@@ -428,7 +428,7 @@ class LineReader {
   const int fd_;
   char * const buf_;
   const int buf_len_;
-  const char *bol_;
+  char *bol_;
   char *eol_;
   const char *eod_;  // End of data in "buf_".
 };
index e8cbc0c..f06bcdd 100644
--- src/utilities.cc
+++ src/utilities.cc
@@ -3,6 +3,9 @@

 #include "utilities.h"

+#include <stdio.h>
+#include <stdlib.h>
+
 #include <signal.h>
 #ifdef HAVE_SYS_TIME_H
 # include <sys/time.h>
